File: libs/src/main/java/org/zsy/libs/framework/app/BaseActivity.java
package org.zsy.libs.framework.app;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class BaseActivity extends AppCompatActivity {

    public final String TAG = getClass().getSimpleName();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityManager.getInstance().addActivity(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        ActivityManager.getInstance().removeActivity(this);
    }
}

File: libs/src/main/java/org/zsy/libs/framework/app/BaseActivity.kt
package org.zsy.libs.framework.app

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import org.zsy.libs.framework.app.ActManager

abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        ActManager.getInstance().addActivity(this)
    }


    override fun onDestroy() {
        super.onDestroy()
        ActManager.getInstance().removeActivity(this)
    }


}
